source/imaer-shared/src/main/java/nl/overheid/aerius/shared/emissions/FarmlandEmissionsCalculator.java

```diff
@@ -23,20 +23,52 @@
 import nl.overheid.aerius.shared.domain.Substance;
 import nl.overheid.aerius.shared.domain.v2.source.FarmlandEmissionSource;
 import nl.overheid.aerius.shared.domain.v2.source.farmland.FarmlandActivity;
+import nl.overheid.aerius.shared.domain.v2.source.farmland.FarmlandGrazingActivity;
 
 public class FarmlandEmissionsCalculator {
 
-  public Map<Substance, Double> calculateEmissions(final FarmlandEmissionSource farmlandSource) {
+  private final FarmlandEmissionFactorSupplier farmlandEmissionFactorSupplier;
+
+  public FarmlandEmissionsCalculator(final FarmlandEmissionFactorSupplier farmlandEmissionFactorSupplier) {
+    this.farmlandEmissionFactorSupplier = farmlandEmissionFactorSupplier;
+  }
+
+  public Map<Substance, Double> updateEmissions(final FarmlandEmissionSource farmlandSource) {
     final Map<Substance, BigDecimal> summed = new EnumMap<>(Substance.class);
     for (final FarmlandActivity activity : farmlandSource.getSubSources()) {
-      final Map<Substance, Double> emissionsForActivity = activity.getEmissions();
-      emissionsForActivity.forEach(
-          (key, value) -> summed.merge(key, BigDecimal.valueOf(value), (v1, v2) -> v1.add(v2)));
+
+      if (activity instanceof FarmlandGrazingActivity) {
+        updateEmissions((FarmlandGrazingActivity) activity, summed);
+      } else {
+        updateEmissions(activity, summed);
+      }
```

Comment:
I'm not sure if this is warranted yet, but ideally we use a visitor pattern here, especially if more of these kinds of `instanceof` cases (will) exist

Reply:
I'm going to add outdoor yards as well, so that seems like the right moment to change this pattern :)